### PR TITLE
Bootstrap project using kubebuilder 3.1.0 

### PR DESCRIPTION
**Pull request by baijum**
Commands used to bootstrap:
```
kubebuilder init --domain="service.binding" --repo="github.com/kubernetes-sigs/service-binding" --owner="The Kubernetes Authors" --project-name="service-binding-operator"
kubebuilder edit --multigroup=true
kubebuilder create api --crd-version="v1" --group="" --version="v1alpha2" --kind="ServiceBinding" --resource --controller
kubebuilder create api --crd-version="v1" --group="" --version="v1alpha2" --kind ClusterApplicationResourceMapping --namespaced=false --resource --controller
make generate
```

### CODE REVIEW

**Comment by sbose78** [comment]
Are these inter-operable with kubebuilder?

File: PROJECT [hunk 1]
```diff
@@ -0,0 +1,27 @@
+domain: service.binding
+layout:
+- go.kubebuilder.io/v3
+multigroup: true
+plugins:
+  manifests.sdk.operatorframework.io/v2: {}
```
Comment (scothis) [inline PROJECT:6]: can we start with vanilla kubebuilder?

Comment (baijum) [reply PROJECT]: Done.

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# More info: https://docs.docker.com/engine/reference/builder/#dockerignore-file
+# Ignore build and test binaries.
+bin/
```
[the remark this answers is not in the document]
Comment (baijum) [reply .dockerignore]: Done.

File: PROJECT [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+repo: github.com/kubernetes-sigs/service-binding
+resources:
+- api:
+    crdVersion: v1
+    namespaced: true
+  controller: true
+  domain: service.binding
+  kind: ServiceBinding
+  path: github.com/kubernetes-sigs/service-binding/apis/v1beta1
+  version: v1beta1
```
[the remark this answers is not in the document]
Comment (baijum) [reply PROJECT]: Done.

**Comment by baijum** [comment]
I have bootstrapped the project using kubebuilder. Force pushed the commit as the old commit is obsolete.

File: PROJECT [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    crdVersion: v1
+    namespaced: true
+  controller: true
+  domain: service.binding
+  kind: ServiceBinding
+  path: github.com/kubernetes-sigs/service-binding/apis/v1alpha2
+  version: v1alpha2
+- api:
+    crdVersion: v1
+    namespaced: true
```
Comment (scothis) [inline PROJECT:18]: should not be namespaced

File: config/rbac/clusterapplicationresourcemapping_editor_role.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# permissions for end users to edit clusterapplicationresourcemappings.
```
Comment (scothis) [inline config/rbac/clusterapplicationresourcemapping_editor_role.yaml:1]: is this file used?

File: config/rbac/clusterapplicationresourcemapping_viewer_role.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# permissions for end users to view clusterapplicationresourcemappings.
```
Comment (scothis) [inline config/rbac/clusterapplicationresourcemapping_viewer_role.yaml:1]: is this file used?

File: config/rbac/servicebinding_editor_role.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# permissions for end users to edit servicebindings.
```
Comment (scothis) [inline config/rbac/servicebinding_editor_role.yaml:1]: is this file used?

File: config/rbac/servicebinding_viewer_role.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# permissions for end users to view servicebindings.
```
Comment (scothis) [inline config/rbac/servicebinding_viewer_role.yaml:1]: is this file used?

File: PROJECT (hunk 4 above)
Comment (baijum) [reply PROJECT]: Fixed. 295e16d4ceb409b484b4899341ed97bd8dc65274

File: config/rbac/clusterapplicationresourcemapping_editor_role.yaml (hunk 5 above)
Comment (baijum) [reply config/rbac/clusterapplicationresourcemapping_editor_role.yaml:1]: The RBAC-related files are auto-generated by kubebuilder. Probably helpful for those who are packing this project (say Helm charts).

Comment (scothis) [reply config/rbac/clusterapplicationresourcemapping_editor_role.yaml:1]: ok. I was a bit surprised to not see it listed in the kustomization.yaml file, even if commented

**Review by scothis — APPROVED** [review]

**Pull request merged**